src/library_syscall.js

```diff
@@ -821,12 +822,17 @@ var SyscallsLibrary = {
         }
       };
     }
+#endif
     for (var i = 0; i < iovcnt; i++) {
       var ptr = {{{ makeGetValue('iov', 'i*8', 'i32') }}};
       var len = {{{ makeGetValue('iov', 'i*8 + 4', 'i32') }}};
+#if USE_PRINTCHARS_FOR_STDOUT == 0  
```

Comment:
This might be nicer to read in positive direction, i.e. `#if USE_PRINTCHARS_FOR_STDOUT` instead of the negative?